pyeapi/api/staticroute.py

```diff
@@ -0,0 +1,302 @@
+# BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY,
+# WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE
+# OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN
+# IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+"""Module for working with EOS static routes
+
+The staticroute resource provides configuration management of static
+route resources on an EOS node. It provides the following class
+implementations:
+
+    * StaticRoute - Configure static routes in EOS
+
+StaticRoute Attributes:
+    ip_dest (string): The ip address of the destination in the
+        form of A.B.C.D/E
+    next_hop (string): The next hop interface or ip address
+    next_hop_ip (string): The next hop address on destination interface
+    distance (int): Administrative distance for this route
+    tag (int): Route tag
+    route_name (string): Route name
+
+Notes:
+    The 'default' prefix function of the 'ip route' command,
+    'default ip route ...', currently equivalent to the 'no ip route ...'
+    command.
+"""
+
+import re
+
+from pyeapi.api import EntityCollection
+
+# Define the regex to match ip route lines (by lines in regex):
+#   'ip route' header
+#   ip_dest
+#   next_hop
+#   next_hop_ip
+#   distance
+#   tag
+#   name
+ROUTES_RE = re.compile(r'(?<=^ip route)'
+                       r' (\d+\.\d+\.\d+\.\d+\/\d+)'
+                       r' (\d+\.\d+\.\d+\.\d+|\S+)'
+                       r'(?: (\d+\.\d+\.\d+\.\d+))?'
+                       r' (\d+)'
+                       r'(?: tag (\d+))?'
+                       r'(?: name (\S+))?', re.M)
+
+# # Define a format for the unique route id
+# # The four parts in order are ip_dest, next_hop, next_hop_ip, and distance
+# ROUTE_ID = "%s--%s--%s--%s"
+
+
+class StaticRoute(EntityCollection):
+    """The StaticRoute class provides a configuration instance
+    for working with static routes
+
+    """
+
+    def __str__(self):
+        return 'StaticRoute'
+
+    def get(self, ip_dest, next_hop, distance, next_hop_ip=None):
+        """Retrieves the ip route information for the route specified
+            by the ip_dest, next_hop, and distance parameters
+
+        Args:
+            ip_dest (string): The ip address of the destination in the
+                form of A.B.C.D/E
+            next_hop (string): The next hop interface or ip address
+            distance (string): Administrative distance for this route
+
+        Returns:
+            dict: An ip route dict object
+
+            If the unique route specified by the ip_dest, next_hop, and
+            distance does not exist, then None is returned.
+        """
+
+        # If distance is None, then set to 1 to match what EOS will
+        # do when distance is not specified.
+        if distance is None:
+            distance = 1
+
+        # Make the unique route_id tuple for the requested route
+        route_id = (ip_dest, next_hop, next_hop_ip, int(distance))
```

Comment:
Something to think about: If I pass in distance='a', your call for int(distance) will throw an error.  You may think about a try and except and raise a ValueError if it can't cast it into an int.